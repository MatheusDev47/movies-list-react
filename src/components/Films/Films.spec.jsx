import { render, screen } from '@testing-library/react';

import { Films } from '.';

const props = {
  films: [
    {
      id: 1,
      title: 'title 01',
      overview: 'overview 01',
      poster_path: '/img/img01.png',
    },
    {
      id: 2,
      title: 'title 02',
      overview: 'overview 02',
      poster_path: '/img/img02.png',
    },
    {
      id: 3,
      title: 'title 03',
      overview: 'overview 03',
      poster_path: '/img/img03.png',
    },
    {
      id: 4,
      title: 'title 04',
      overview: 'overview 04',
      poster_path: '/img/img04.png',
    },
  ],
};

describe('<Films />', () => {
  it('Should render movies', () => {
    render(<Films {...props} />);

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(4);
    expect(screen.getAllByText(/overview/i)).toHaveLength(4);
    expect(screen.getByRole('img', { name: /title 01/i })).toHaveAttribute(
      'src',
      'https://image.tmdb.org/t/p/w500/img/img01.png',
    );
  });

  it('Should not render movies when no movies exist', () => {
    render(<Films />);

    expect(screen.queryByRole('heading', { name: /title 02/i })).not.toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const { container } = render(<Films {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
