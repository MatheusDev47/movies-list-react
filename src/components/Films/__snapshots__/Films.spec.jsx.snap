// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Films /> Should match snapshot 1`] = `
<div
  class="content"
>
  <div
    class="card"
  >
    <img
      alt="title 01"
      src="https://image.tmdb.org/t/p/w500/img/img01.png"
    />
    <div
      class="film-content"
    >
      <h2>
        title 01
      </h2>
      <p>
        overview 01
      </p>
    </div>
  </div>
  <div
    class="card"
  >
    <img
      alt="title 02"
      src="https://image.tmdb.org/t/p/w500/img/img02.png"
    />
    <div
      class="film-content"
    >
      <h2>
        title 02
      </h2>
      <p>
        overview 02
      </p>
    </div>
  </div>
  <div
    class="card"
  >
    <img
      alt="title 03"
      src="https://image.tmdb.org/t/p/w500/img/img03.png"
    />
    <div
      class="film-content"
    >
      <h2>
        title 03
      </h2>
      <p>
        overview 03
      </p>
    </div>
  </div>
  <div
    class="card"
  >
    <img
      alt="title 04"
      src="https://image.tmdb.org/t/p/w500/img/img04.png"
    />
    <div
      class="film-content"
    >
      <h2>
        title 04
      </h2>
      <p>
        overview 04
      </p>
    </div>
  </div>
</div>
`;
